### ISSUE TITLE
Integrate reward/instructor implementation into trainer_rm

### ISSUE THREAD

**Issue by andreaskoepf** [LAION-AI/Open-Assistant#2309]
Currently we have two different reward model training code branches, e.g. model/model_training/trainer_rm.py vs. model/reward/instructor/trainer.py which causes some confusion.

We need to decide what should happen with the old code in `model/reward/instructor/trainer.py`. If this folder contains valuable code to train a reward model (with a different loss) it should be integrated into the newer model/model_training/trainer_rm.py code, e.g. to be activated by a configuration option (e.g. type of loss function or type of reward model).

**Comment by CloseChoice** [comment]
There is already an issue and a linked PR regarding this issue:
https://github.com/LAION-AI/Open-Assistant/pull/2289
Would be good if somebody could check if that is fine. I mentioned the things where I am unsure about there.

**Comment by CloseChoice** [comment]
Since the linked issue is closed and the corresponding PR is merged I close this
